src/ResourceManager/Compute/Commands.Compute/VirtualMachineSizes/GetAzureVMSizeCommand.cs

```diff
@@ -78,30 +79,30 @@ public override void ExecuteCmdlet()
 
             ExecuteClientAction(() =>
             {
-                VirtualMachineSizeListResponse result = null;
+                AzureOperationResponse<IPage<VirtualMachineSize>> result = null;
 
                 if (!string.IsNullOrEmpty(this.VMName))
                 {
-                    result = this.VirtualMachineClient.ListAvailableSizes(
+                    result = this.VirtualMachineClient.ListAvailableSizesWithHttpMessagesAsync(
                         this.ResourceGroupName,
-                        this.VMName);
+                        this.VMName).GetAwaiter().GetResult();
                 }
                 else if (!string.IsNullOrEmpty(this.AvailabilitySetName))
                 {
-                    result = this.AvailabilitySetClient.ListAvailableSizes(
+                    result = this.AvailabilitySetClient.ListAvailableSizesWithHttpMessagesAsync(
                         this.ResourceGroupName,
-                        this.AvailabilitySetName);
+                        this.AvailabilitySetName).GetAwaiter().GetResult();
                 }
                 else
                 {
-                    result = this.VirtualMachineSizeClient.List(this.Location.Canonicalize());
+                    result = this.VirtualMachineSizeClient.ListWithHttpMessagesAsync(this.Location.Canonicalize()).GetAwaiter().GetResult();
                 }
 
                 List<PSVirtualMachineSize> psResultList = new List<PSVirtualMachineSize>();
-                foreach (var item in result.VirtualMachineSizes)
+                foreach (var item in result.Body)
```

Comment:
OK, but ```List``` doesn't contain the operation info that we want.

Will it be feasible to return a named collection in the async method? [feature request]